packages/packages/v1-adapters/src/hooks/use-is-preview-mode.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+import { editModeChangeEvent, getCurrentEditMode, listenTo } from '../';
+
+export default function useIsPreviewMode() {
+	const [ isPreview, setIsPreview ] = useState( () => isPreviewMode() );
+
+	useEffect( () => {
+		const updateState = () => setIsPreview( isPreviewMode() );
+
+		const cleanup = listenTo(
+			editModeChangeEvent(),
+			updateState,
+		);
+
+		return cleanup;
+	}, [] );
+
+	return isPreview;
+}
```

Comment:
```suggestion
export default function useListenTo( events, checkState ) {
	const [ state, setState ] = useState( checkState );

	useEffect( () => {
		const updateState = () => setState( checkState() );

		const cleanup = listenTo(
			events,
			updateState,
		);

		return cleanup;
	}, [] );

	return state;
}

useListenTo( editModeChangeEvent(), () => getCurrentEditMode() === 'preview' )
```

Reply:
Great idea!
changed it a little bit, WDYT?